src/profile.c

```diff
@@ -316,7 +316,6 @@ PRINT_PROFILE_FUNC(printUnionIt) {
   char *unionTypeStr;
   switch (ui->type) {
   case QN_GEO : unionTypeStr = "GEO"; break;
-  case QN_GEOMETRY : unionTypeStr = "GEOSHAPE"; break;
```

Comment:
why is this removed? did we remove from the enum?

Reply:
There is no scenario of `GEOMETRY` specific union, so it was never should’ve been added here. It should be considered an error (he faked by the `default`)

Reply:
should it be removed from the `query_node.h`? or we are considered about potential backward compatiblility or something?

Reply:
No. This switch doesn't have to handle all possible query node types; only those that may be used as a union type (passed to its constructor) are required.

Why do we have this mechanism? Why not have a different enum or a generic "to_string" helper? Greate questions for another PR :)